Add in some PhaseII Code.

trmazi opened the pull request:
Not all of this code is working! GFDM V8 does, and Hello! Pop'n does. they're both VERY messy but they should be able to merge.

trmazi committed:
add updated constants, need changing.

trmazi committed:
add backend folders for DM, GF, and Hello! Popn Music. Not linked to anything

GFDM V6-V4 WILL NOT work. they're broken

trmazi committed:
add updated read and services

THESE WILL PROBABLY BREAK YOUR SERVER!!!!!! THEY ARE MISSING FILES

trmazi committed:
Merge branch 'DragonMinded:trunk' into trunk

trmazi committed:
clean up the code a weeee bit

trmazi committed:
Merge branch 'trunk' of https://github.com/trmazi/bemaniutils into trunk

DragonMinded commented on .gitignore:
```diff
@@ -15,3 +15,4 @@ bemani.egg-info/
 server/
 emblems
 .vscode/
+.DS_Store
```
https://i.imgur.com/t9Kkt8h.jpg

DragonMinded commented on bemani/backend/dm/base.py:
```diff
@@ -0,0 +1,354 @@
+
+class DrummaniaBase(CoreHandler, CardManagerHandler, Base):
+    """
+    Trying to work on GFDM right now. starting off strong with DM :)
```
Please remove irrelevant comments, they don't belong as docstrings. If you want to instead document what is expected to work and what isn't, this is an okay place to do that.

DragonMinded commented on bemani/backend/dm/base.py:
```diff
@@ -0,0 +1,354 @@
+
+    def __init__(self, data: Data, config: Dict[str, Any], model: Model) -> None:
+        super().__init__(data, config, model)
+        if model.rev == 'X':
```
Does DM have omnimix or is this copypasta?

DragonMinded commented on bemani/backend/dm/base.py:
```diff
@@ -0,0 +1,354 @@
+                    attempts[songid][songchart] = {
+                        'total': 0,
+                        'clears': 0,
+                        'average': 0,
```
Do we not update the average anywhere after here? Is this copied from some buggy code elsewhere?

DragonMinded commented on bemani/backend/dm/base.py:
```diff
@@ -0,0 +1,354 @@
+        history.replace_int('combo', combo)
+
+        #write our skill points
+        print(skill_perc)
```
Please go through and remove all debug prints from the code before it can be merged.

DragonMinded commented on bemani/backend/dm/base.py:
```diff
@@ -0,0 +1,354 @@
+        scoredata.replace_int('combo', max(scoredata.get_int('combo'), combo))
+        history.replace_int('combo', combo)
+
+        #write our skill points
```
Fairly sure this will result in a lint error for no space between # and write. Please make sure before you submit in the future that you run `./verifylint` as well as `./veritytyping` to catch these sorts of things.

DragonMinded commented on bemani/backend/dm/base.py:
```diff
@@ -0,0 +1,354 @@
+        history.replace_int('clear', clear)
+
+        #did we cheat?
+        #scoredata.replace_int('autoclear', max(scoredata.get_int('autoclear'), autoclear))
```
Shouldn't this function just reject the score outright if its a cheat? Why is this here?

trmazi replied:
the game sends that if you use auto pedal
it doesn't read it anywhere though, so i chose not to record it

DragonMinded replied:
Okay, then we should probably get rid of this section as well as the autoclear parameter.

DragonMinded commented on bemani/backend/dm/base.py:
```diff
@@ -0,0 +1,354 @@
+        # same user and song/chart. So, bump the timestamp by one second and retry
+        # well past the maximum number of songs.
+        now = Time.now()
+        for bump in range(10):
```
This just unconditionally adds 1 + 2 + 3 + 4 + 5 + 6 + 7 + 8 + 9 + 10 to the timestamp. You aren't checking for anything here?

trmazi replied:
the game does not send a timestamp :(
if there's a cleaner way to do this, please let me know

DragonMinded replied:
Several games don't send timestamps. That's not what this comment is pointing out. This comment is pointing out that you are literally just adding in a loop. You don't actually handle multiple attempts of the same song.

DragonMinded commented on bemani/backend/dm/base.py:
```diff
@@ -0,0 +1,354 @@
+        
+
+        # Write the new score back
+        self.data.local.music.put_score(
```
If you need to modify the timestamp, it should be done in a try/catch loop with this function which will fail when inserting a duplicate score. Was this tested? I presume not because I don't think you wrote any traffic tests for this.

trmazi replied:
it was tested, and works most of the time (not good)

DragonMinded replied:
Yeah, this code doesn't do what you think it does, it is literally not trying to insert scores and fixing duplicates at all.

DragonMinded commented on bemani/backend/dm/dmv4.py:
```diff
@@ -0,0 +1,521 @@
+    name = "Drummania V4"
+    version = VersionConstants.DRUMMANIA_V4
+
+    GAME_LIMITED_LOCKED = 1
```
This entire file looks like a rough copy-pasta of SDVX, including packets and blocks and limited stuff. I doubt DMV4 works the same way. Honestly if you need to check in stubs, this should be an empty file. Preferrably it shouldn't even be included in the review. I'm going to skip reviewing this whole file because it looks like its not mergeable as-is.

trmazi replied:
it's copy pasta, yes. forgot to remove this from the public fork..... oops

DragonMinded commented on bemani/backend/dm/dmv5.py:
```diff
@@ -0,0 +1,521 @@
+    name = "Drummania V5"
+    version = VersionConstants.DRUMMANIA_V5
+
+    GAME_LIMITED_LOCKED = 1
```
Same comment here, get rid of completely useless code that just copies SDVX please.

trmazi replied:
it's copy pasta, yes. forgot to remove this from the public fork..... oops

DragonMinded commented on bemani/backend/dm/dmv6.py:
```diff
@@ -0,0 +1,521 @@
+        return {
+            'bools': [
+                {
+                    'name': 'Disable Online Matching',
```
Same comment here... drop this file.

trmazi replied:
it's copy pasta, yes. forgot to remove this from the public fork..... oops

DragonMinded commented on bemani/backend/dm/dmv7.py:
```diff
@@ -0,0 +1,2512 @@
+    GAME_LIMITED_LOCKED = 1
+    GAME_LIMITED_UNLOCKED = 2
+
+    GAME_CURRENCY_PACKETS = 0
```
And here...

trmazi replied:
shouldn't be copypasta?? but maybe sent the wrong thing

DragonMinded commented on bemani/backend/dm/dmv7.py:
```diff
@@ -0,0 +1,2512 @@
+
+        refid = data.child_value('refid')
+        name = data.child_value('name')
+        gdp = 0
```
Why is all this here instead of just in the dictionary below? No need for so many tempvars.

trmazi replied:
Was cleaned in a later version

DragonMinded commented on bemani/backend/dm/dmv7.py:
```diff
@@ -0,0 +1,2512 @@
+        userid = self.data.remote.user.from_refid(self.game, self.version, refid)
+        profile = self.get_profile(userid)
+        
+
```
W
H
Y
?

DragonMinded commented on bemani/backend/dm/dmv7.py:
```diff
@@ -0,0 +1,2512 @@
+        history = Node.void('history')
+        battledata.add_child(history)
+
+        round_before = Node.void('round')
```
Use a loop for all this, why is it copy-pasta'd 10 times?

trmazi replied:
loops are hard, this was also like one of the first things i wrote. fixing this soon though.

DragonMinded commented on bemani/backend/dm/dmv7.py:
```diff
@@ -0,0 +1,2512 @@
+
+
+
+        #no idea what this does lol
```
What are these comments?

trmazi replied:
because my dumb ass sent the wrong file. this is from before i had any idea on how to load the scores

DragonMinded commented on bemani/backend/dm/dmv7.py:
```diff
@@ -0,0 +1,2512 @@
+        player.add_child(Node.string('my_rival_id', "0"))
+        player.add_child(Node.u8('mode', profile.get_int('playmode')))
+
+        #oh god oh fuck
```
>:|

DragonMinded commented on bemani/backend/dm/dmv8.py:
```diff
@@ -0,0 +1,2503 @@
+    name = "Drummania V8"
+    version = VersionConstants.DRUMMANIA_V8
+
+    GAME_LIMITED_LOCKED = 1
```
I assume that GFDMv8 is working? Why is there so much SDVX copypasta here still? Did you run lint on this even one time?

DragonMinded commented on bemani/backend/dm/dmv8.py:
```diff
@@ -0,0 +1,2503 @@
+    def previous_version(self) -> Optional[DrummaniaBase]:
+        return DrummaniaV7(self.data, self.config, self.model)
+
+    def db_to_game_chart(self, db_chart: int) -> int:
```
Is there a corresponding game_to_db_chart? I don't see any of the constants being used above. How do you convert between game constants and DB constants? Am I missing something?

DragonMinded commented on bemani/backend/dm/dmv8.py:
```diff
@@ -0,0 +1,2503 @@
+        info = Node.void('info')
+        root.add_child(info)
+
+        info.add_child(Node.string('message', "Brought to you by PhaseII! Hosted on phaseii.iidxfan.xyz. Login today!"))
```
Yeah no, what the fuck? Did you check or test any of this code before putting up a PR?

trmazi replied:
i DEF forgot to remove that, sorry. 
this is going to be replaced with an arcade webui setting soon.

DragonMinded commented on bemani/backend/dm/dmv8.py:
```diff
@@ -0,0 +1,2503 @@
+
+        refid = data.child_value('refid')
+        name = data.child_value('name')
+        gdp = 0
```
Same comment here, just move these into the dict below??

trmazi replied:
same response above.

DragonMinded commented on bemani/backend/dm/dmv8.py:
```diff
@@ -0,0 +1,2503 @@
+        history = Node.void('history')
+        battledata.add_child(history)
+
+        round_before = Node.void('round')
```
None of this is filled in? Does DMv8 work? Is it considered finished? Why is there copypasta for miles instead of a dang loop?

DragonMinded commented on bemani/backend/dm/base.py:
```diff
@@ -0,0 +1,354 @@
+        """
+        return None
+    
+    def get_profile_by_refid(self, refid: Optional[str]) -> Optional[Node]:
```
It seems that this, new_profile_by_refid, put_profile_by_refid and get_clear_rates are all completely unused. Why are they included in this file?

trmazi replied:
because i forgot to send the right file

DragonMinded commented on bemani/backend/gf/base.py:
```diff
@@ -0,0 +1,351 @@
+
+class GuitarFreaksBase(CoreHandler, CardManagerHandler, Base):
+    """
+    Trying to work on GFDM right now. now it's time to play the guitar rhythmically :)
```
Again, please don't pollute docstrings like this.

DragonMinded commented on bemani/backend/gf/base.py:
```diff
@@ -0,0 +1,351 @@
+        history.replace_int('clear', clear)
+
+        #did we cheat?
+        #scoredata.replace_int('autoclear', max(scoredata.get_int('autoclear'), autoclear))
```
Same comment as DM here.

DragonMinded commented on bemani/backend/gf/base.py:
```diff
@@ -0,0 +1,351 @@
+        # since we don't allow two of the same attempt at the same time for the
+        # same user and song/chart. So, bump the timestamp by one second and retry
+        # well past the maximum number of songs.
+        now = Time.now()
```
Also same comment, this code does literally nothing other than change every single timestamp by 55 seconds.

DragonMinded commented on bemani/backend/gf/gfv8.py:
```diff
@@ -0,0 +1,2657 @@
+    GAME_LIMITED_LOCKED = 1
+    GAME_LIMITED_UNLOCKED = 2
+
+    GAME_CURRENCY_PACKETS = 0
```
Looks like SDVX copypasta here too? I skipped reviewing gfv4-v7 because I assume you did the same thing with them as you did with DM.

DragonMinded commented on bemani/backend/gf/gfv8.py:
```diff
@@ -0,0 +1,2657 @@
+    def previous_version(self) -> Optional[GuitarFreaksBase]:
+        return GuitarFreaksV7(self.data, self.config, self.model)
+
+    def db_to_game_chart(self, db_chart: int) -> int:
```
Where's game_to_db_chart? Where is this used?

DragonMinded commented on bemani/backend/gf/gfv8.py:
```diff
@@ -0,0 +1,2657 @@
+            self.CHART_TYPE_OPEN_EXTREME: self.GAME_CHART_TYPE_OPEN_EXTREME,
+        }[db_chart]
+
+    def handle_shopinfo_regist_request(self, request: Node) -> Node:
```
Is this used?

DragonMinded commented on bemani/backend/gf/gfv8.py:
```diff
@@ -0,0 +1,2657 @@
+        info = Node.void('info')
+        root.add_child(info)
+
+        info.add_child(Node.string('message', "Brought to you by PhaseII! Hosted on phaseii.iidxfan.xyz. Login today!"))
```
Dude really? This is about as bad as bundle sites advertising in the zip file where you downloaded somebody else's software.

DragonMinded commented on bemani/backend/gf/gfv8.py:
```diff
@@ -0,0 +1,2657 @@
+        # Now, reload and format the profile, looking up the has old version flag
+        profile = self.get_profile(userid)
+
+        # oldversion = self.previous_version()
```
Why is this commented out?

DragonMinded commented on bemani/backend/gf/gfv8.py:
```diff
@@ -0,0 +1,2657 @@
+        history = Node.void('history')
+        battledata.add_child(history)
+
+        round_before = Node.void('round')
```
Again with the completely unpopulated chunks, why is this not handled and not in a loop? You have access to the score history, this doesn't look done at all.

DragonMinded commented on bemani/backend/gf/gfv8.py:
```diff
@@ -0,0 +1,2657 @@
+        playedmode = modedata.attribute('mode')
+
+        if playedmode == "standard" :
+            lastmode = 1
```
Use defines instead of magic values here.

DragonMinded commented on bemani/backend/gf/gfv8.py:
```diff
@@ -0,0 +1,2657 @@
+
+        standard = Node.void('standard')
+
+        easy = 0
```
Aren't these exactly the same as the constants above? Why are these here?

DragonMinded commented on bemani/common/constants.py:
```diff
@@ -1,4 +1,5 @@
 class GameConstants:
+    ROAD_FIGHTERS = 'roadfighters'
```
Why is this here? This is not related to the PR.

DragonMinded commented on bemani/common/constants.py:
```diff
@@ -8,11 +9,39 @@ class GameConstants:
+    GUITARFREAKS = 'gf'
+    GITADORA = 'gtd'
+    POPN_HELLO = 'hpm'
+    BEATSTREAM = 'bst'
```
Same with all of these. Please only include relevant changes.

DragonMinded commented on bemani/common/constants.py:
```diff
@@ -164,6 +250,28 @@ class DBConstants:
     DDR_RANK_AA_PLUS = 650
     DDR_RANK_AAA = 700
 
+    DANCE_EVOLUTION_HALO_NONE = 100
```
Same with all of this.

DragonMinded commented on bemani/utils/services.py:
```diff
@@ -9,12 +9,19 @@
 from bemani.backend.sdvx import SoundVoltexFactory
 from bemani.backend.reflec import ReflecBeatFactory
 from bemani.backend.museca import MusecaFactory
+#from bemani.backend.roadfighters import RoadFightersFactory
```
Get rid of this commented out stuff.

DragonMinded commented on bemani/utils/services.py:
```diff
@@ -151,6 +160,18 @@ def register_games() -> None:
         SoundVoltexFactory.register_all()
     if config.get('support', {}).get(GameConstants.MUSECA, False):
         MusecaFactory.register_all()
+    #if config.get('support', {}).get(GameConstants.ROAD_FIGHTERS, False):
```
Same here.

DragonMinded commented on bemani/backend/gf/base.py:
```diff
@@ -0,0 +1,351 @@
+        profile = self.get_profile(userid)
+        return self.format_profile(userid, profile)
+
+    def put_profile_by_refid(self, refid: Optional[str], request: Node) -> None:
```
Same comment here for all _by_refid functions, they appear completely unused. Is this true?

DragonMinded commented on bemani/backend/popnhell/base.py:
```diff
@@ -0,0 +1,98 @@
+        """
+        return None
+
+    def new_profile_by_refid(self, refid: Optional[str], name: Optional[str], locid: Optional[int]) -> Node:
```
Same here, these don't appear to be used by the implementation at all.

DragonMinded commented on bemani/backend/popnhell/popnhello.py:
```diff
@@ -0,0 +1,137 @@
+        flag = Node.void('flag')
+        root.add_child(flag)
+
+        flag.set_attribute(id, '1')
```
Was this even tested? None of these variables are defined, this does nothing but crash?

trmazi replied:
sent the wrong file. this will def break things.

DragonMinded commented on bemani/backend/popnhell/popnhello.py:
```diff
@@ -0,0 +1,137 @@
+    def handle_game_new_request(self, request: Node) -> Node:
+        #game_new
+        root = Node.void('game')
+        name = "0"
```
Why are these here, and why not in the validated dict? Why do they even have to be in the default profile at all? get_int will return a "0" for anything not in the profile already, this is entirely useless. Also, I realize that that's the case where you did this for gfdm as well. Best to remove it.

DragonMinded commented on bemani/backend/popnhell/popnhello.py:
```diff
@@ -0,0 +1,137 @@
+
+        return root
+
+    def handle_game_load_m_request(self, request: Node) -> Node:
```
Is this in hello popn? Where are scores loaded and saved? This looks like it boots the game and that's it.

DragonMinded commented on bemani/utils/read.py:
```diff
@@ -3482,23 +5383,24 @@ def import_music_db(self, songs: List[Dict[str, Any]]) -> None:
             songs = ddr.lookup(args.server, args.token)
         else:
-            if args.version == '16':
+            if args.version >= '16':
```
Unrelated change?

DragonMinded commented on bemani/utils/read.py:
```diff
@@ -3482,23 +5383,24 @@ def import_music_db(self, songs: List[Dict[str, Any]]) -> None:
                     'provide either a --bin, --tsv or a --server and --token option!'
                 )
             iidx.import_music_db(songs)
+            iidx.import_qpros(qpros)
```
Unrelated?

DragonMinded commented on bemani/utils/read.py:
```diff
@@ -3473,6 +5313,67 @@ def import_music_db(self, songs: List[Dict[str, Any]]) -> None:
+    elif args.series == GameConstants.GUITARFREAKS:
+        gf = ImportGuitarFreaks(config, args.version, args.no_combine, args.update)
+        if args.tsv is not None:
+            # Special case for Jubeat, grab the title/artist metadata that was
```
Copypasta?

trmazi replied:
GFDM also needs a tsv for artist and song data.
forgot to change that 😨

DragonMinded commented on bemani/utils/read.py:
```diff
@@ -3473,6 +5313,67 @@ def import_music_db(self, songs: List[Dict[str, Any]]) -> None:
+            # hand-populated since its not in the music DB.
+            gf.import_metadata(args.tsv)
+        else:
+            # Normal case, doing a music DB or emblem import.
```
This looks completely copypasta, and there's no tsv for gfdm songs, why is a lot of this even here?

trmazi replied:
again, GFDM requires a TSV, like jubeat/iidx

DragonMinded commented on bemani/utils/read.py:
```diff
@@ -12,7 +12,7 @@
 import xml.etree.ElementTree as ET
 from sqlalchemy import create_engine  # type: ignore
-from sqlalchemy.engine import CursorResult  # type: ignore
+from sqlalchemy.engine.result import ResultProxy  # type: ignore
```
This reverts the change for sqlalchemy compatibility?

DragonMinded reviewed (changes requested):
Thanks for putting this up! However, this is so far from mergeable that I suspect you'll have to spend a lot of time working on it. There's several files where you made them executable for some reason, read.py is so littered with unrelated changes that I gave up reviewing it, there's a massive amount of copypasta, there seems to be so many lint errors and I see absolutely no tests whatsoever which would have caught some bugs in the code. None of the stuff that's listed as working is complete. You also are trying to advertise your own server instance in the code which is tacky at best.

trmazi committed:
the finished hello! popn music code

I cleaned up read.py, and attached the tsv file as well

trmazi committed:
oops

trmazi committed:
stub out and remove useless GFDMs

trmazi committed:
remove dumb message

trmazi committed:
Fix errors in Hello! Pop'n Music

After I did some more testing (when it was not midnight) I found some more errors in the code. these have since been fixed.

DragonMinded commented:
This is clearly going nowhere and was a complete waste of my time to even review. Closing because it is cluttering PRs.
